<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Playfair</class>
 <widget class="QWidget" name="Playfair">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>975</width>
    <height>502</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Шифр Плейфера</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QFrame" name="LeftFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QFrame" name="AlphabetOptionsFrame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="AlphabetLabel">
           <property name="text">
            <string>Алфавит</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="AlphabetHeightSpin">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="AlphabetWidthSpin">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="EnglishAlphabetButton">
           <property name="text">
            <string>Английский</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="RussianAlphabetButton">
           <property name="text">
            <string>Русский</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="HorizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="AlphabetTable">
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="verticalHeaderMinimumSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>24</number>
        </attribute>
        <row>
         <property name="text">
          <string>1</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>2</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>3</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>4</string>
         </property>
        </row>
        <row>
         <property name="text">
          <string>5</string>
         </property>
        </row>
        <column>
         <property name="text">
          <string>1</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>2</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>3</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>4</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>5</string>
         </property>
        </column>
        <item row="0" column="0">
         <property name="text">
          <string>A</string>
         </property>
        </item>
        <item row="0" column="1">
         <property name="text">
          <string>B</string>
         </property>
        </item>
        <item row="0" column="2">
         <property name="text">
          <string>C</string>
         </property>
        </item>
        <item row="0" column="3">
         <property name="text">
          <string>D</string>
         </property>
        </item>
        <item row="0" column="4">
         <property name="text">
          <string>E</string>
         </property>
        </item>
        <item row="1" column="0">
         <property name="text">
          <string>F</string>
         </property>
        </item>
        <item row="1" column="1">
         <property name="text">
          <string>G</string>
         </property>
        </item>
        <item row="1" column="2">
         <property name="text">
          <string>H</string>
         </property>
        </item>
        <item row="1" column="3">
         <property name="text">
          <string>IJ</string>
         </property>
        </item>
        <item row="1" column="4">
         <property name="text">
          <string>K</string>
         </property>
        </item>
        <item row="2" column="0">
         <property name="text">
          <string>L</string>
         </property>
        </item>
        <item row="2" column="1">
         <property name="text">
          <string>M</string>
         </property>
        </item>
        <item row="2" column="2">
         <property name="text">
          <string>N</string>
         </property>
        </item>
        <item row="2" column="3">
         <property name="text">
          <string>O</string>
         </property>
        </item>
        <item row="2" column="4">
         <property name="text">
          <string>P</string>
         </property>
        </item>
        <item row="3" column="0">
         <property name="text">
          <string>Q</string>
         </property>
        </item>
        <item row="3" column="1">
         <property name="text">
          <string>R</string>
         </property>
        </item>
        <item row="3" column="2">
         <property name="text">
          <string>S</string>
         </property>
        </item>
        <item row="3" column="3">
         <property name="text">
          <string>T</string>
         </property>
        </item>
        <item row="3" column="4">
         <property name="text">
          <string>U</string>
         </property>
        </item>
        <item row="4" column="0">
         <property name="text">
          <string>V</string>
         </property>
        </item>
        <item row="4" column="1">
         <property name="text">
          <string>W</string>
         </property>
        </item>
        <item row="4" column="2">
         <property name="text">
          <string>X</string>
         </property>
        </item>
        <item row="4" column="3">
         <property name="text">
          <string>Y</string>
         </property>
        </item>
        <item row="4" column="4">
         <property name="text">
          <string>Z</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="KeyFrame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="KeyLabel">
           <property name="text">
            <string>Ключ</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="KeyEdit"/>
         </item>
         <item>
          <widget class="QLabel" name="StubLabel">
           <property name="text">
            <string>Заглушка</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="StubEdit">
           <property name="maximumSize">
            <size>
             <width>24</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>X</string>
           </property>
           <property name="maxLength">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="EncodingMatrixLabel">
        <property name="text">
         <string>Матрица шифрования</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="EncodingMatrixTable">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="verticalHeaderMinimumSectionSize">
         <number>24</number>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>24</number>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="DigramsLabel">
        <property name="text">
         <string>Биграммы</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="DigramsEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QFrame" name="RightFrame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="1">
       <widget class="QPushButton" name="EncodeButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Зашифровать</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QPushButton" name="DecodeButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Расшифровать</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="MessageLabel">
        <property name="text">
         <string>Исходный текст</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="CodeLabel">
        <property name="text">
         <string>Криптограмма</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QPlainTextEdit" name="CodeEdit"/>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QPlainTextEdit" name="MessageEdit"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
